pkg/morph/client/client.go

```diff
@@ -367,23 +367,50 @@ func (c *Client) TestInvoke(contract util.Uint160, method string, args ...any) (
 
 // TestInvokeIterator is the same [Client.TestInvoke] but expands an iterator placed
 // on the stack. Returned values are the values an iterator provides.
-func (c *Client) TestInvokeIterator(contract util.Uint160, method string, args ...any) (res []stackitem.Item, err error) {
+// If prefetchElements > 0, that many elements are tried to be placed on stack without
+// additional network communication (without the iterator expansion).
+func (c *Client) TestInvokeIterator(contract util.Uint160, method string, prefetchElements int, args ...any) (res []stackitem.Item, err error) {
 	c.switchLock.RLock()
 	defer c.switchLock.RUnlock()
 
 	if c.inactive {
 		return nil, ErrConnectionLost
 	}
 
-	sid, iter, err := unwrap.SessionIterator(c.rpcActor.Call(contract, method, args...))
-	if err != nil {
-		return nil, fmt.Errorf("iterator expansion: %w", err)
+	var sid uuid.UUID
+	var iter result.Iterator
+	items := make([]stackitem.Item, 0)
+
+	if prefetchElements > 0 {
+		script, err := smartcontract.CreateCallAndPrefetchIteratorScript(contract, method, prefetchElements, args...)
+		if err != nil {
+			return nil, fmt.Errorf("building prefetching script: %w", err)
+		}
+
+		resInvoke, err := c.rpcActor.Run(script)
```

Comment:
Why not `unwrap.ArrayAndSessionIterator(c.rpcActor.Run(script))`?

Reply:
cause `ArrayAndSessionIterator` says `ArrayAndSessionIterator expects correct execution (HALT state)`?

Reply:
oh, i see, it has the check anyway. hm. i feel like it was not like that when i read "expects correct execution" last time. i would say such words require me to be sure it is true on my side

Reply:
So, can we drop some lines?

Reply:
> can we drop some lines?

if you do confirm that `ArrayAndSessionIterator expects correct execution` _does not_ mean i should ensure it on my side -- yes. fixed

Reply:
All of `unwrap.*` is made to check for the execution result and it's specified.

Reply:
this specification is ambiguous

Reply:
Well, https://pkg.go.dev/github.com/nspcc-dev/neo-go/pkg/rpcclient/unwrap

> These functions will check for error, check for VM state ...

> ArrayAndSessionIterator expects correct execution (HALT state) ...

How can it be misinterpreted?

Reply:
I would use "expect" word (and seems like neofs docs usually go it this way) for arguments requirements that a caller should do